### ISSUE TITLE
Collection state management hook

### ISSUE THREAD

**Issue by mimarz** [equinor/design-system#1186]
blocks #605 

We are currently maintaining two duplicate implementations of managing states when navigating a collection of items and which one is in focus/current. With navigation drawer also needing the same logic its counterintuitive to have yet another separate implementation of this.

Suggest creating or using an existing state hook for managing a collection of items and which is in focus as well as provide correct aria-props that should be applied

Affected components: 
* Navigation Drawer (not started)
* Menu
* Tabs

Inspiration:
* https://react-spectrum.adobe.com/react-stately/useTreeState.html
* https://github.com/sparksuite/react-accessible-dropdown-menu-hook
